### DISCUSSION TITLE
Optuna Bayesian Optimization giving the exact same score for different hyperparameters

### DISCUSSION THREAD

**Discussion by pedroheck** [optuna/optuna#4061] [Q&A]
### Don't use GitHub Issues to ask support questions.

Hey there! I'm trying for the first time to use a Bayesian Optimization algorithm to find the best hyperparameters for an SVM model in python. Since I'm quite a newbie, all I did was copy the code from [this article](https://towardsdatascience.com/grid-search-vs-random-search-vs-bayesian-optimization-2e68f57c3c46#:%7E:text=Unlike%20the%20grid%20search%20and,is%20determined%20by%20the%20user). It's working fine, except for one thing: in many iterations, the score given is exactly the same, even though the hyperparameters change by even orders of magnitude. For example:

```
[I 2022-10-09 14:53:41,257] A new study created in memory with name: no-name-9d9006dd-76ef-436c-a8a4-c02487429bf0

[I 2022-10-09 14:54:23,540] Trial 0 finished with value: 0.5007206305675235 and parameters: {'C': 0.0010793486838214896, 'gamma': 4334.896592673167, 'kernel': 'rbf'}. Best is trial 0 with value: 0.5007206305675235.

[I 2022-10-09 14:54:53,157] Trial 1 finished with value: 0.5007206305675235 and parameters: {'C': 1.2589532558683528e-08, 'gamma': 0.007300535499260778, 'kernel': 'rbf'}. Best is trial 0 with value: 0.5007206305675235.

[I 2022-10-09 14:56:52,134] Trial 2 finished with value: 0.5062454889593887 and parameters: {'C': 21035238.23328419, 'gamma': 2745656433.2504015, 'kernel': 'rbf'}. Best is trial 2 with value: 0.5062454889593887.

[I 2022-10-09 14:58:46,419] Trial 3 finished with value: 0.6222678036787913 and parameters: {'C': 397647751.01648784, 'gamma': 25.448563305149033, 'kernel': 'rbf'}. Best is trial 3 with value: 0.6222678036787913.

[I 2022-10-09 14:59:17,078] Trial 4 finished with value: 0.5007206305675235 and parameters: {'C': 0.005009772855375645, 'gamma': 3.540327068803278e-05, 'kernel': 'rbf'}. Best is trial 3 with value: 0.6222678036787913.

[I 2022-10-09 14:59:33,359] Trial 5 finished with value: 0.8397792508810221 and parameters: {'C': 24217.278857801153, 'gamma': 2.3042682362831503e-07, 'kernel': 'rbf'}. Best is trial 5 with value: 0.8397792508810221.

[I 2022-10-09 15:00:15,466] Trial 6 finished with value: 0.5007206305675235 and parameters: {'C': 1.3694061073764904e-07, 'gamma': 2926028.7075905013, 'kernel': 'rbf'}. Best is trial 5 with value: 0.8397792508810221.

[I 2022-10-09 15:00:52,915] Trial 7 finished with value: 0.5007206305675235 and parameters: {'C': 5.608467784347845e-10, 'gamma': 0.01723070885334284, 'kernel': 'rbf'}. Best is trial 5 with value: 0.8397792508810221.
```

Look how 0.5007206305675235 appears repeatedly. What's even weirder is that this exact score number appears if I change from F1-score to accuracy score, which makes even less sense given their entirely different natures.

My code for the Bayesian Optimization is as follows:

```
import optuna 
from optuna.samplers import TPESampler
from sklearn.model_selection import cross_val_score
import time


def objective(trial):
    """return the accuracy"""

    # search space for svm
    C = trial.suggest_float('C', 1e-10, 1e10, log=True)
    gamma = trial.suggest_float('gamma', 1e-10, 1e10, log=True)
    kernel = trial.suggest_categorical('kernel', ['rbf'])
    # degree = trial.suggest_int('degree', low=0, high=7, step=1)

    # model
    model = svm.SVC(C=C, gamma=gamma, kernel=kernel, random_state=42)
    
    score =  cross_val_score(estimator=model, 
        X=X_train_sacaled, 
        y=y_train_res, 
        scoring='accuracy',
        cv=5,
        n_jobs=-1
    ).mean()

    return score

# create a study (aim to maximize score)
study = optuna.create_study(sampler=TPESampler(), direction='maximize')

# perform hyperparamter tuning (while timing the process)
time_start = time.time()
study.optimize(objective, n_trials=100)
time_bayesian = time.time() - time_start

# store result in a data frame 
values_bayesian = [100, study.best_trial.number, study.best_trial.value, time_bayesian]
columns = ['Number of iterations', 'Iteration Number of Optimal Hyperparamters', 'Score', 'Time Elapsed (s)']
results_bayesian = pd.DataFrame([values_bayesian], columns = columns)
```

Notice that I'm using the accuracy score here, but I've tried with the F1 score and some numbers were still repeating (including 0.5007206305675235). Does anyone have any idea why this might be happening? I legit have no clue.

Thank you in advance!

**Comment by toshihikoyanase** [comment]
This seems to be a question. Let me transfer it to the GitHub Discussions.

**Comment by nzw0301** [comment]
Depending on the problem, it might happen, especially when the problem is simple, for example, a small dataset or a simple model.
